### PR TITLE
feat: allow download charts as image,pdf

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: [#11529](https://github.com/lightdash/lightdash/issues/11529)

### Description:

Allows downloading charts as image,pdf,json.etc - uses the same component as the Explorer

Demo: 

https://github.com/user-attachments/assets/53d131c8-f5e2-4a4a-aab1-771f60dcd61b

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/ChartDownload.tsx [hunk 1]
```diff
@@ -215,18 +209,14 @@ const ChartDownloadOptions: React.FC<DownloadOptions> = ({
...
-                    size="xs"
-                    ml="auto"
-                    leftIcon={<MantineIcon icon={IconDownload} />}
-                    onClick={onDownload}
-                >
-                    {' '}
-                    Download
-                </Button>
-            )}
+            <Button
```
Comment (joaoviana) [inline packages/frontend/src/components/ChartDownload.tsx:212]: we would never reach this statement. It's moved down below before rendering this component.

File: packages/frontend/src/features/sqlRunner/components/Download/ChartDownload.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                                label={capitalize(DownloadFileType.CSV)}
+                                my="xs"
+                            />
+                            <Radio
+                                value={DownloadFileType.IMAGE}
+                                label={capitalize(DownloadFileType.IMAGE)}
+                            />
+                        </Radio.Group>
+                        {downloadFormat === DownloadFileType.IMAGE && (
+                            <ChartDownloadOptions
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/Download/ChartDownload.tsx:64]: use the same chart download options when choosing `image` download format.

**Review by rephus — APPROVED** [review]
it works great. 

Can we remo the JSON option though ? 

it looks like it is the results in JSON format, but it looks more like the echart configa, and also, it is not an "image" format...
Instead, we could add a proper JSON format , next to CVS and image, to download the results

image

**Comment by joaoviana** [comment]
> it works great.
> 
> Can we remo the JSON option though ?
> 
> it looks like it is the results in JSON format, but it looks more like the echart configa, and also, it is not an "image" format... Instead, we could add a proper JSON format , next to CVS and image, to download the results (on a separate ticket perhaps)
> 
> image

Great point @rephus , removed that option on sql runner charts 👍 

We can add the option later in a separate label when we feel the need from user feedback 🙌 


Screenshot 2024-09-23 at 15 01 16
Removed in sql runner 🆙

**Pull request merged**